ihatemoney/models.py

```diff
@@ -263,6 +271,37 @@ def get_pretty_bills(self, export_format="json"):
             )
         return pretty_bills
 
+    def switch_currency(self, new_currency):
+        # Update converted currency
+        if new_currency != self.default_currency:
+            if (
+                new_currency == CurrencyConverter.no_currency
+                and self.has_multiple_currencies()
+            ):
+                raise ValueError(f"Can't unset currency of project {self.id}")
+            for bill in self.get_bills_unordered():
+
+                if new_currency == CurrencyConverter.no_currency:
+                    # Use old currency to flatten all amount before stripping
+                    bill.converted_amount = CurrencyConverter().exchange_currency(
+                        bill.amount, bill.original_currency, self.default_currency
+                    )
+                    # Strip currency
+                    bill.amount = bill.converted_amount
+                    bill.original_currency = CurrencyConverter.no_currency
```

Comment:
I don't get why we need to convert the amount here.  When `new_currency == XXX`, we just need to remove the currency indication on all bills. The amount doesn't need to be converted (because we checked that all bills have the same currency). Also, in principle, we should never touch `bill.amount`.

There is this edge-case that is not completely clear:

- all bills have original_currency=EUR
- current default currency is USD
- new default currency is XXX

In that case, I would still not convert the bills to USD before removing the currency: it's not useful and modifies the original amount for no good reason.

So in the end this case can be simplified to simply:
```
# We are removing the currency, and we already checked that all bills had the same currency:
# it means that we can simply strip the currency without converting the amounts.
# We basically ignore the current default_currency.
if new_currency == CurrencyConverter.no_currency:
    # Reset converted amount in case it was different from the original amount
    bill.converted_amount = bill.amount
    # Strip currency
    bill.original_currency = CurrencyConverter.no_currency
```

Reply:
Hmm, I am puzzled by the actual code, but I guess that's because it was done when we still allowed the switch to 'XXX' with multiple used currencies.
In the current scenario, I agree that we could simply strip the currency, full stop.

I will write a test first for this.

Reply:
Now there is a test. This fails (obviously), so please tell me if it fits the use case.

Reply:
Cool, the test looks good!